src/utils/searchAndSort.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  SearchNodePayload,
+  SearchSortOrder,
+  isAndNodePayload,
+  isFieldNodePayload,
+  isNotNodePayload,
+  isOrNodePayload,
+} from 'src/types/Search';
+
+export type SearchOrderConfig = {
+  locale: string | null;
+  sortOrder: SearchSortOrder;
+  numberFields?: string[];
+};
+
+export const splitTrigrams = (value: string): string[] => {
+  const trigrams = [];
+  let position;
+
+  // Remove non-alphanumeric characters
+  const _value = value.replace(/[^0-9a-z]/gi, ' ').replace(/\s+/, ' ');
+
+  // Split into words, pad each word with spaces
+  const words = _value.split(' ').map((word) => ` ${word} `);
+
+  for (const word of words) {
+    position = 3;
+    // For some reason the postgres implementation also includes the first 2 characters of each word
+    trigrams.push(word.substring(0, 2));
+
+    // Split the words into trigrams
+    for (let i = 0; i < word.length; i += 1) {
+      if (position <= word.length) {
+        trigrams.push(word.substring(i, position).toLowerCase());
+        position += 1;
+      }
+    }
+  }
+
+  return trigrams;
+};
+
+const searchConditionMet = <T extends Record<string, unknown>>(result: T, condition: SearchNodePayload): boolean => {
```

Comment:
This is very cool 😎 ... a tad bit of a reservation that search should be done on the BE side maybe. Have a consistent search behavior across pages. I understand for MVP this is our decision,  the endpoints are not search table implementations.

Reply:
I agree 100%. That's why I tried to make it as close to the original API as possible, in the hopes that we can just swap it over easily when that day comes.

Reply:
Agree on that. This is very cool and should be pretty consistent 🙌